src/Extensions/SiteTreeSubsites.php

```diff
@@ -524,23 +515,26 @@ public function augmentSyncLinkTracking()
      */
     public function augmentValidURLSegment()
     {
-        // If this page is being filtered in the current subsite, then no custom validation query is required.
-        $subsite = Subsite::$force_subsite ?: SubsiteState::singleton()->getSubsiteId();
-        if (empty($this->owner->SubsiteID) || $subsite == $this->owner->SubsiteID) {
-            return null;
-        }
-
-        // Backup forced subsite
-        $prevForceSubsite = Subsite::$force_subsite;
-        Subsite::$force_subsite = $this->owner->SubsiteID;
-
-        // Repeat validation in the correct subsite
-        $isValid = $this->owner->validURLSegment();
-
-        // Restore
-        Subsite::$force_subsite = $prevForceSubsite;
-
-        return (bool)$isValid;
+        return SubsiteState::singleton()->withState(function (SubsiteState $newState) {
```

Comment:
This entire block seems wrong

Have a look at other implementations of withState() on extensions within the subsites module e.g. `FileSubsites::canEdit()`

```php
        return SubsiteState::singleton()->withState(function (SubsiteState $newState) use ($member) {
            $newState->setSubsiteId($this->owner->SubsiteID);

            return $this->owner->getPermissionChecker()->canEdit($this->owner->ID, $member);
        });
```

You're supposed to call `$newState->setSubsiteId($subsiteID);` on the first line of the callback

Reply:
I'm not sure that I should set subsite ID and after that do the following checking 
```
$subsite = $newState->getSubsiteId() ?: SubsiteState::singleton()->getSubsiteId();
if (empty($this->owner->SubsiteID) || $subsite == $this->owner->SubsiteID) {
...
```
I think this `$subsite == $this->owner->SubsiteID` should return `true` is I set ID.

Reply:
I *think* this block is now supposed to be

```php
  return SubsiteState::singleton()->withState(function (SubsiteState $newState) {
            $newState->setSubsiteId($this->owner->SubsiteID);
            return (bool) $this->owner->validURLSegment();
  });
```

The whole point is to get a return value from `$this->owner->validURLSegment()` running in the context of the "current" subsite being force to $this->owner->SubsiteID.  All of the deprecated 'force_subsite' logic should be replaced by the withState()